### DISCUSSION TITLE
Is there a built-in sorting strategy for virtualized grids in `@dnd-kit/sortable`?

### DISCUSSION THREAD

**Discussion by kombucha** [clauderic/dnd-kit#411] [Q&A] [answered]
Hi 👋 

I'm currently using [`react-sortable-hoc`](https://github.com/clauderic/react-sortable-hoc) and I'm evaluating how we could migrate to `dnd-kit` instead. 
In my application, I have a virtualized, sortable, grid of items which is powered by `react-sortable-hoc` and [`react-virtual`](https://github.com/tannerlinsley/react-virtual) (I'm actually virtualizing pre-chunked "rows of items", so no virtualization in the x axis).

I checked the docs (notably [sorting strategies](https://docs.dndkit.com/presets/sortable#sorting-strategies)) and the examples storybook and unless I'm mistaken virtualized grids don't seem to be supported out of the box.

Did I miss anything? If not, would you have some pointers on how this could be implemented in userland?

Thank you!

**Comment by clauderic** [comment, marked as answer]
That's right, virtualized grids aren't currently supported by any of the sorting strategies. It's been a while since I looked at this, but I believe the `rectSortingStrategy` strategy needs all elements to be mounted to work properly.

Having said that, you could fairly easily look at the source code of  [`react-sortable-hoc`](https://github.com/clauderic/react-sortable-hoc)'s grid sorting logic and port it over as a custom sorting strategy for @dnd-kit:

https://github.com/clauderic/react-sortable-hoc/blob/master/src/SortableContainer/index.js#L711-L788

This isn't a feature I need so I won't be prioritizing it myself for the time being. Would happily accept a PR request for @dnd-kit if anyone tackles this.

**Reply by michaeled** [reply]
Hey @clauderic,

Do you know if this feature is still in the backlog? If so, I'm happy to assist with the development and send a PR.

**Reply by tomasmenezes** [reply]
@clauderic @michaeled Would definitely appreciate support for this feature

**Reply by hsavit1** [reply]
@michaeled any updates ?

**Reply by hsavit1** [reply]
@clauderic will the new dnd kit implementation support this feature?

**Comment by bnjmnt4n** [comment]
Actually, I have a virtualized grid implementation which works pretty well with `@dnd-kit/sortable`. I'm using `rectSortingStrategy`, and there doesn't seem to be any issues even without all the items being rendered. The only thing I had to do was to ensure that the item currently being dragged was rendered even when it's outside of the visible range. The code is using both a `DragOverlay` and including the item as a sortable, and perhaps the only issue is that if a currently dragged item is outside of the visible range per the virtualization, and the drag is cancelled, the `DragOverlay` cannot perform a transition since the rendered dragged item will no longer be rendered when there is no active drag.

This can probably be worked around by adding a timeout equivalent to the transition duration to ensure that the sortable item is visible until the `DragOverlay` animates back, but I wanted to see if there are any other issues/suggestions with this implementation. In particular, are there any issues @clauderic had perhaps noted in the past, which required specific handling?

**Reply by taiwanhua** [reply]
@bnjmnt4n  Hi, May I see your code about virtualized grid implementation?
I am struggling with this,
need help~~

**Reply by taiwanhua** [reply]
In the end I had to add pagination to let the browser render only part of the item

**Comment by jonathanrstern** [comment]
Any updates on this?

I'm using rectSortingStrategy and haven't figured out a virtualized grid implementation.

@bnjmnt4n
